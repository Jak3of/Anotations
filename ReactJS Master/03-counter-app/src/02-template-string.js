export function getSaludo(nombre) {
	return 'Hola ' + nombre
}

// console.log(`este es un texto: ${getSaludo(nombre)}`)


export const getUser = () => ({
	uid: 'ABC123',
	username: 'Nelson'
})




export const getUsuarioActivo = (nombre) => ({
	uid: 'ABC123',
	username: nombre
})