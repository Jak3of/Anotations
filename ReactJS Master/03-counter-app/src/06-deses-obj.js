

export const usContext = ({ clave , nombre , edad , rango = 'Capitan' }) => {

    return {
        nombreClave: clave,
        petName : nombre,
        petRango : rango,
        anios : edad,
        latlng : {
            lat : 14.1232,
            lng : -12.3232
        }
    }
}

export const retornaArreglo = () => {

    return ['ABC', 123];
}