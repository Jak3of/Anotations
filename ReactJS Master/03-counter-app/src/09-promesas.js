import { getHeroesById } from './08-imp-exp';

export const getHeroesByIdAsync = ( id ) => {

    return new Promise( ( resolve, reject ) => {

        setTimeout( () => {

            const hero = getHeroesById( id );
            if ( hero ) {
                resolve( hero );
            } else {
                reject( `No existe un héroe con el id ${ id }` );
            }
        }, 1000 );
    })
}