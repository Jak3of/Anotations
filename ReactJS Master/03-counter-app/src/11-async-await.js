
export const getImagen = async() => {
    try {
    const apikey = 'bfwqsSYhjQ1N2sZNgKRzX21K6L7Z9qbz';
    const res = await fetch(`https://api.giphy.com/v1/gifs/random?api_key=${apikey}`);
    const {data} = await res.json();

    const {url} = data.images.original;

    return url;
    } catch (error) {
        console.error(error);
        return 'No se encontro la imagen';
    }

}
