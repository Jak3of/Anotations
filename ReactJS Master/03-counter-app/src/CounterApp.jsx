
import PropTypes from 'prop-types';
import { useState } from 'react';



export function CounterApp( {value} ) {


	const [seg, setSeg] = useState(value);
	const [min, setMin] = useState(0);
	const [hor, setHor] = useState(0);

	const handleAdd = () =>	setSeg(seg + 1)


	const handleSub = () => setSeg(seg - 1)
	
	const handleReset = () =>{
		
		setSeg(value);
		setMin(0);
		setHor(0);
	}
	
	// setTimeout(() => {
		
	// 	setSeg(prevSeg => prevSeg + 1);
	
	// }, 1000);
	
	if (seg === 60) {
		setSeg(0);
		setMin( min + 1 );
		
		}
	
	if (min === 60) {
		setMin(0);
		setHor( hor + 1 );
	}
	
	if (seg === -1 && min > 0) {
		setSeg(59);
		setMin(min - 1);
		if (min === -1 && hor > 0) {
			setHor(hor - 1);
		}
	} else if (seg === -1 && min === 0 && hor === 0) {
		setSeg(0);
		setMin(0);
		setHor(0);
	}
	console.log('render');

	return (
		<>
            <h1>CounterApp </h1>
			<h3 style={{ textAlign: 'center' }}>{seg}</h3>
			<h2 style={{ textAlign: 'center' }}>{hor}:{min}:{seg}</h2>

			<button className="btn" onClick={ handleAdd } >+1</button>
			<button className="btn" onClick={ handleSub } >-1</button>
			<button className="btn" aria-label="btn-reset" onClick={ handleReset } >Reset</button>
		</>
		)
}

CounterApp.propTypes = {
	value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
	value: 0
}


