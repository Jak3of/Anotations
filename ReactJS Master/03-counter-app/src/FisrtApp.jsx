
import PropTypes from 'prop-types';

export default function FirstApp({title , subTitle, name}) {
	return (<>
		<h1>{title}</h1>
		<p>{subTitle}</p>
		<p>{name}</p>
	</>)
}



FirstApp.propTypes = {
	title: PropTypes.string.isRequired,
	subTitle: PropTypes.string,
	name: PropTypes.string

}

FirstApp.defaultProps = {
	subTitle: 'No hay subtitulo',
	title: 'No hay titulo',
	name: 'Nelson'
}


