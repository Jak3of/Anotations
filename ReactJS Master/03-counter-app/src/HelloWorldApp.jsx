


const newMessage = [1,2,3,4,5,6,7,8,9];

export function App() {


	return (<>
			<code>{ newMessage}</code>
			<p> Hola soy un parrafor</p>
		</>)
}


