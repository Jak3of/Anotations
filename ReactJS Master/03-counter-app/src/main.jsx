import React from 'react';
import ReactDOM from 'react-dom/client';
import { App } from './HelloWorldApp'
import FirstApp from './FisrtApp'
import { CounterApp } from './CounterApp'
import './style.css'


ReactDOM.createRoot( document.getElementById('root')).render(
		<React.StrictMode>
			<App/>
			<FirstApp title="Hola soy Vegeta"/>
			<CounterApp value={10}/>
			
		</React.StrictMode>
);

