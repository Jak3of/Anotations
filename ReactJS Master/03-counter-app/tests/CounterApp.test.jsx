import { render , screen , fireEvent } from '@testing-library/react';

import { CounterApp } from '../src/CounterApp';

import { describe, test , expect } from '@jest/globals';


describe('Pr_Test en <CounterApp/>',() => {

    const initialValue = 100;

    test('debe de hacer match con el snapshot',() =>{

        const {container} = render(<CounterApp value={ initialValue}/>);


        expect(container).toMatchSnapshot();

    })

    test('debe de mostrar el valor por defecto de 100',() =>{
        render(<CounterApp value={ 100 }/>);
        expect(screen.getByText(100)).toBeTruthy();

        // expect(screen.getByRole('heading',{level: 3}).innerHTML).toBe('100')




    })


    test('debe de incrementar con el boton +1',() =>{

        render(<CounterApp value={ initialValue }/>);
        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByText('101')).toBeTruthy();
    })

    test('debe de decrementar con el boton -1',() =>{

        render(<CounterApp value={ initialValue }/>);
        fireEvent.click(screen.getByText('-1'));
        expect(screen.getByText('99')).toBeTruthy();
    })

    test('debe de hacer reset con el boton Reset',() =>{

        render(<CounterApp value={ initialValue }/>);
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByRole('button',{name:'btn-reset'}));
        expect(screen.getByText(100)).toBeTruthy();
    })




})
