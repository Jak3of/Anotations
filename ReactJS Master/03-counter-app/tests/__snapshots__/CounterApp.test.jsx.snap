// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pr_Test en <CounterApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp 
  </h1>
  <h3
    style="text-align: center;"
  >
    100
  </h3>
  <h2
    style="text-align: center;"
  >
    0
    :
    0
    :
    100
  </h2>
  <button
    class="btn"
  >
    +1
  </button>
  <button
    class="btn"
  >
    -1
  </button>
  <button
    aria-label="btn-reset"
    class="btn"
  >
    Reset
  </button>
</div>
`;
